.github/workflows/test.yml

```
@@ -0,0 +1,138 @@
+  push:
+    branches: [ main ]
+
+jobs:
+  # Run unit tests (no API calls)
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up Go
+      uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)